vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -89,10 +90,7 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 
 	dial := config.Dial
 	if dial == nil {
-		dial = (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-		}).DialContext
+		dial = libgonetwork.DefaultClientDialContext()
```

Comment:
this is what we want, compare with https://github.com/openshift/origin/pull/25773